src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -525,25 +375,26 @@ private IReadOnlyList<ITemplateInfo> TemplatesToDisplayInfoAbout
             {
                 IEnumerable<ITemplateInfo> templateList;
 
-                if (UnambiguousTemplateGroupToUse != null && UnambiguousTemplateGroupToUse.Count > 0)
+                if (_templateResolutionResult.UnambiguousTemplateGroupToUse != null && _templateResolutionResult.UnambiguousTemplateGroupToUse.Count > 0)
                 {
-                    templateList = UnambiguousTemplateGroupToUse.Select(x => x.Info);
+                    templateList = _templateResolutionResult.UnambiguousTemplateGroupToUse.Select(x => x.Info);
                 }
-                else if (!string.IsNullOrEmpty(TemplateName) && _matchedTemplatesWithSecondaryMatchInfo != null && _matchedTemplatesWithSecondaryMatchInfo.Count > 0)
+                else if (!string.IsNullOrEmpty(TemplateName) && _templateResolutionResult.MatchedTemplatesWithSecondaryMatchInfo != null && _templateResolutionResult.MatchedTemplatesWithSecondaryMatchInfo.Count > 0)
```

Comment:
Should we expose a property on the resolution result that is the result of the last two parts of the expression? Something like HasMatchedTemplatesWithSecondaryMatchInfo?